proxy_network.c

```diff
@@ -1510,6 +1510,9 @@ static void proxy_backend_handler(const int fd, const short which, void *arg) {
     // Still pending requests to read or write.
     if (!STAILQ_EMPTY(&be->io_head)) {
         flags |= EV_READ; // FIXME (v2): might not be necessary here, but ensures we get a disconnect event.
+        if (!be->can_write) {
+            flags |= EV_WRITE;
+        }
```

Comment:
Bravo! 

Is this the cause of write halt we saw in tcpdump?

Reply:
Among other things yes. This had different symptoms depending on what you were doing. These bugs all worked together to cause problems.

Reply:
What I will likely do as soon as I have better test tools:
- split to-write/to-read IO queues
- split to-read and to-write event handlers (maybe). I have them merged because libevent is inefficient but I don't like the bug that this caused.